iree/compiler/Dialect/HAL/Target/VMLA/test/linking.mlir

```diff
@@ -232,3 +232,87 @@ module {
 // CHECK-NEXT:    }
 // CHECK-NEXT:    return
 // CHECK-NEXT:  }
+
+// -----
+
+module {
+  hal.executable @dispatch_0 attributes {sym_visibility = "private"} {
+    hal.interface @legacy_io {
+      hal.interface.binding @arg0, set=0, binding=0, type="StorageBuffer", access="Read"
+      hal.interface.binding @arg1, set=0, binding=1, type="StorageBuffer", access="Read"
+      hal.interface.binding @ret0, set=0, binding=2, type="StorageBuffer", access="Write|Discard"
+    }
+    hal.executable.target @vmla, filter="vmla" {
+      hal.executable.entry_point @dispatch_0 attributes {interface = @legacy_io, ordinal = 0 : i32, signature = (tensor<1x1xf32>, tensor<1x1xf32>) -> tensor<1x1xf32>}
+      module {
+        vm.module @module {
+          vm.func @dispatch_0(%arg0: !vm.ref<!vmla.interface>, %arg1: i32, %arg2: i32, %arg3: i32) {
+            vm.return
+          }
+          vm.export @dispatch_0
+        }
+      }
+    }
+  }
+  hal.executable @dispatch_1 attributes {sym_visibility = "private"} {
+    hal.interface @legacy_io attributes {push_constants = 2 : i32} {
+      hal.interface.binding @arg0, set=0, binding=0, type="StorageBuffer", access="Read"
+      hal.interface.binding @arg1, set=0, binding=1, type="StorageBuffer", access="Read"
+      hal.interface.binding @ret0, set=0, binding=2, type="StorageBuffer", access="Write|Discard"
+    }
+    hal.executable.target @vmla, filter="vmla" {
+      hal.executable.entry_point @dispatch_1 attributes {interface = @legacy_io, ordinal = 0 : i32, signature = (tensor<1x1xf32>, tensor<1x1xf32>) -> tensor<1x1xf32>}
+      module {
+        vm.module @module {
+          vm.func @dispatch_1(%arg0: !vm.ref<!vmla.interface>, %arg1: i32, %arg2: i32, %arg3: i32) {
+            vm.return
+          }
+          vm.export @dispatch_1
+        }
+      }
+    }
+  }
+  hal.executable @dispatch_2 attributes {sym_visibility = "private"} {
+    hal.interface @legacy_io attributes {push_constants = 2 : i32} {
+      hal.interface.binding @arg0, set=0, binding=0, type="StorageBuffer", access="Read"
+      hal.interface.binding @arg1, set=0, binding=1, type="StorageBuffer", access="Read"
+      hal.interface.binding @ret0, set=0, binding=2, type="StorageBuffer", access="Write|Discard"
+    }
+    hal.executable.target @vmla, filter="vmla" {
+      hal.executable.entry_point @dispatch_2 attributes {interface = @legacy_io, ordinal = 0 : i32, signature = (tensor<1x1xf32>, tensor<1x1xf32>) -> tensor<1x1xf32>}
+      module {
+        vm.module @module {
```

Comment:
you can (I believe) remove the entire module {} for these as you aren't checking them - just the targets and entry points should be enough to test that you end up with the new interfaces in your linked executable

Reply:
Hmm.. I can try again after meeting. This seemed minimal, to get around some DCE?

Reply:
https://github.com/google/iree/blob/98753255d76dfb5ae809812843364102d2f6beec/iree/compiler/Dialect/HAL/Target/VMLA/VMLATarget.cpp#L94-L98 is looking for `vm.module` ops, and I get a pretty confusing 
```
LLVM ERROR: Building op `vm.module` but it isn't registered in this MLIRContext
```
error if I omit them.

Reply:
I can remove the _contents_ of the `vm.module` ops though.